[Feature Request]: Fix progress-report GitHub CI check

### Feature description

The progress-report PR check is failing with various unrelated errors (see log attached below).

However, running the commands required for this by hand in a development environment shows that the test should be passing.  Investigate and fix the check such that all CI passes

### Describe alternatives you've considered

_No response_

### Anything else?

error log:
```sh
> VENV=$VIRTUAL_ENV make report
KFP clone version: 1.8.4
Creating Python virtual environment ...
WARNING: Package(s) not found: kfp
Installing KFP SDK 1.8.14 ...
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [60 lines of output]
      /tmp/pip-build-env-wq6eef2_/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
      !!
      
              ********************************************************************************
              The license_file parameter is deprecated, use license_files instead.
      
              By 2023-Oct-30, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
      
              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************
      
      !!
        parsed = self.parsers.get(option_name, lambda x: x)(value)
      running egg_info
      writing lib3/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib3/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib3/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/home/runner/work/data-science-pipelines/data-science-pipelines/temp/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/runner/work/data-science-pipelines/data-science-pipelines/temp/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/runner/work/data-science-pipelines/data-science-pipelines/temp/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-wq6eef2_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-wq6eef2_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-wq6eef2_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 271, in <module>
          self.filelist.extend(build_ext.get_source_files())
        File "<string>", line 201, in get_source_files
        File "/usr/lib/python3.10/distutils/cmd.py", line 103, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
WARNING: Package(s) not found: pytest
Installing pytest ...
KFP Python SDK version(s):

SUCCESS: add_pod_env.py
SUCCESS: artifact_passing_using_volume.py
SUCCESS: basic.py
FAILURE: basic_no_decorator.py
SUCCESS: coin.py
FAILURE: compose.py
SUCCESS: default_value.py
SUCCESS: input_artifact_raw_value.py
SUCCESS: loop_over_lightweight_output.py
SUCCESS: opsgroups.py
SUCCESS: parallelfor_item_argument_resolving.py
SUCCESS: parallelfor_pipeline_param_in_items_resolving.py
SUCCESS: param_op_transform.py
SUCCESS: param_substitutions.py
SUCCESS: pipelineparams.py
SUCCESS: recursive_do_while.py
SUCCESS: recursive_while.py
SUCCESS: resourceop_basic.py
SUCCESS: sidecar.py
SUCCESS: timeout.py
SUCCESS: two_step.py
FAILURE: uri_artifacts.py
SUCCESS: volume.py
SUCCESS: volume_snapshotop_rokurl.py
SUCCESS: volume_snapshotop_sequential.py
SUCCESS: volumeop_basic.py
SUCCESS: volumeop_dag.py
SUCCESS: volumeop_parallel.py
SUCCESS: volumeop_sequential.py
SUCCESS: withitem_basic.py
SUCCESS: withitem_nested.py
SUCCESS: withparam_global.py
SUCCESS: withparam_global_dict.py
SUCCESS: withparam_output.py
SUCCESS: withparam_output_dict.py

Compilation status for testdata DSL scripts:

  Success: 32
  Failure: 3
  Total:   35

Overall success rate: 32/35 = 91%

Compilation status report:   sdk/python/tests/test_kfp_samples_report.txt
Accumulated compiler logs:   temp/test_kfp_samples_output.txt
Compiled Tekton YAML files:  temp/tekton_compiler_output/

NOTE: Please update this script to install required Python modules:
 - 'yaml'

This compilation report (left) differs from the previous report (right):

FAILURE: basic_no_decorator.py	      |	SUCCESS: basic_no_decorator.py
FAILURE: compose.py		      |	SUCCESS: compose.py

make: *** [Makefile:98: report] Error 1
Error: Process completed with exit code 2.
```

--- a/.github/workflows/kfp-tekton-unittests.yml
+++ b/.github/workflows/kfp-tekton-unittests.yml
@@ -58,7 +58,7 @@ jobs:
     - name: Install dependencies
       run: python -m pip install -e sdk/python
     - name: "Progress report on compiling KFP DSL test scripts"
-      run: VENV=$VIRTUAL_ENV make report
+      run: VENV=$VIRTUAL_ENV make report PRINT_ERRORS="TRUE"
   python-lint:
     runs-on: ubuntu-latest
     steps:

--- a/sdk/python/tests/test_kfp_samples.sh
+++ b/sdk/python/tests/test_kfp_samples.sh
@@ -76,6 +76,8 @@ TEKTON_COMPILED_YAML_DIR="${TEMP_DIR}/tekton_compiler_output"
 COMPILER_OUTPUTS_FILE="${TEMP_DIR}/test_kfp_samples_output.txt"
 CONFIG_FILE="${PROJECT_DIR}/sdk/python/tests/config.yaml"
 REPLACE_EXCEPTIONS="FALSE" # "TRUE" | "FALSE"
+CYTHON_PACKAGE="cython<3.0.0"
+PYYAML_PACKAGE="pyyaml==5.4.1"
 
 # print KFP SDK and GIT versions (might be different)
 #echo "KFP_GIT_VERSION=${KFP_GIT_VERSION}"
@@ -121,6 +123,13 @@ if [ ! -d "${VENV_DIR}" ]; then
   pip install -q --upgrade pip
 fi
 source "${VENV_DIR}/bin/activate"
+
+# Check if the necessary packages are installed
+if ! (pip show "$CYTHON_PACKAGE" | grep Version) || ! (pip show "$PYYAML_PACKAGE" | grep Version); then
+  echo "Installing necessary packages..."
+  # Install the packages
+  pip install "$CYTHON_PACKAGE" && pip install --no-build-isolation "$PYYAML_PACKAGE"
+fi
 
 # install KFP with the desired KFP SDK version (unless already installed)
 if ! (pip show "kfp" | grep Version | grep -q "${KFP_SDK_VERSION}"); then

--- a/sdk/python/tests/test_kfp_samples.sh
+++ b/sdk/python/tests/test_kfp_samples.sh
@@ -124,12 +124,11 @@ if [ ! -d "${VENV_DIR}" ]; then
 fi
 source "${VENV_DIR}/bin/activate"
 
-# Check if the necessary packages are installed
-if ! (pip show "$CYTHON_PACKAGE" | grep Version) || ! (pip show "$PYYAML_PACKAGE" | grep Version); then
-  echo "Installing necessary packages..."
-  # Install the packages
-  pip install "$CYTHON_PACKAGE" && pip install --no-build-isolation "$PYYAML_PACKAGE"
-fi
+# create a constraint file that limits the Cython version to one that should work
+echo "$CYTHON_PACKAGE" > /tmp/constraint.txt
+
+# install PyYAML itself
+PIP_CONSTRAINT=/tmp/constraint.txt pip install "${PYYAML_PACKAGE}"
 
 # install KFP with the desired KFP SDK version (unless already installed)
 if ! (pip show "kfp" | grep Version | grep -q "${KFP_SDK_VERSION}"); then

--- a/sdk/python/tests/test_kfp_samples.sh
+++ b/sdk/python/tests/test_kfp_samples.sh
@@ -148,6 +148,10 @@ if ! (pip show "pytest" | grep -q Version); then
   pip install -q pytest
 fi
 
+# WORKAROUND - See https://stackoverflow.com/a/76177575
+pip uninstall -y urllib3 requests-toolbelt
+pip install urllib3==1.26.15 requests-toolbelt==0.10.1
+
 # install 3rd party dependencies required for certain pipeline samples
 if [[ "${ALL_SAMPLES}" == "TRUE" ]]; then
   echo "Installing 3rd-party dependencies ..."

